### PR TITLE
e2e: Ensure interchain workflow coverage for the P-Chain

### PR DESCRIPTION
**Pull request by marun**
## Why this should be merged

Fulfills one of the requirements for #1547 (migration of Kurtosis tests)

## How this works

- ensures coverage of interchain transfer for the P-Chain equivalent to and extending from avalanche-testing's [C-Chain Atomic Workflow test](https://github.com/ava-labs/avalanche-testing/blob/master/tests/virtuouscchain/cchain_atomic_workflow.go)

## How this was tested

CI

## TODO

- [x] Merge the parent PR (#1941)

### CODE REVIEW

File: tests/e2e/faultinjection/duplicate_node_id.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		ginkgo.By("checking that the second new node fails to become healthy within timeout")
+		ctx, cancel := context.WithTimeout(context.Background(), e2e.DefaultNodeStartTimeout)
+		defer cancel()
+		err := node2.WaitForHealthy(ctx)
+		require.ErrorIs(err, context.DeadlineExceeded)
+
+		ginkgo.By("stopping the first new node")
+		require.NoError(node1.Stop())
+
+		ginkgo.By("checking that the second new node becomes healthy within timeout")
```
Comment (hexfusion) [inline tests/e2e/faultinjection/duplicate_node_id.go:51]:
```suggestion
		ginkgo.By("checking that the second new node becomes healthy before timeout")
```

**Review by hexfusion — APPROVED** [review]

**Review by danlaine — APPROVED** [review]

File: tests/e2e/p/interchain_workflow.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+			// Validation duration doesn't actually matter to this
+			// test - it is only ensuring that adding a validator
+			// doesn't break interchain transfer.
+			endTime := startTime.Add(30 * time.Second)
+
+			rewardKey, err := factory.NewPrivateKey()
+			require.NoError(err)
+
+			delegationPercent := 0.10 // 10%
+			delegationFee := uint32(reward.PercentDenominator * delegationPercent)
```
Comment (StephenButtolph) [inline tests/e2e/p/interchain_workflow.go:110]:
```suggestion
			const (
				delegationPercent = 0.10 // 10%
				delegationFee     = reward.PercentDenominator * delegationPercent
			)
```

File: tests/e2e/p/interchain_workflow.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("checking that the recipient address has received imported funds on the C-Chain")
+		e2e.Eventually(func() bool {
+			balance, err := ethClient.BalanceAt(e2e.DefaultContext(), recipientEthAddress, nil)
+			require.NoError(err)
+			return balance.Cmp(big.NewInt(0)) > 0
+		}, e2e.DefaultTimeout, e2e.DefaultPollingInterval, "failed to see recipient address funded before timeout")
```
Comment (StephenButtolph) [inline tests/e2e/p/interchain_workflow.go:213]: Because this `ethClient` is talking to the same node that the `cWallet` is talking to - do we need this `Eventually`? I would have thought that this should always pass first try.

File: tests/e2e/e2e.go [hunk 4]
```diff
@@ -59,6 +59,10 @@ const (
 	// checks. Useful for speeding up iteration during test
 	// development.
 	SkipBootstrapChecksEnvName = "E2E_SKIP_BOOTSTRAP_CHECKS"
+
+	// Start time must be a minimum of 15s ahead of the current time
+	// or validator addition will fail.
+	DefaultValidatorStartTimeDiff = 20 * time.Second
```
Comment (abi87) [inline tests/e2e/e2e.go:65]: nit: comment and value are different (15 and 20 seconds).
also nit: I'd rewrite the min start time as  `txexecutor.SyncBond.Add(5*time.Second)` i.e. as the require minimum star time to let a tx be accepted in mempool (see [here](https://github.com/ava-labs/avalanchego/blob/dev/vms/platformvm/block/builder/builder.go#L250)) plus a buffer

File: tests/e2e/p/interchain_workflow.go [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+			// test - it is only ensuring that adding a validator
+			// doesn't break interchain transfer.
+			endTime := startTime.Add(30 * time.Second)
+
+			rewardKey, err := factory.NewPrivateKey()
+			require.NoError(err)
+
+			const (
+				delegationPercent = 0.10 // 10%
+				delegationFee     = reward.PercentDenominator * delegationPercent
```
Comment (abi87) [inline tests/e2e/p/interchain_workflow.go:111]: I'd rename this similaly to what we have done in [staking rewards test addition](https://github.com/ava-labs/avalanchego/pull/1767)

File: tests/e2e/p/interchain_workflow.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+			)
+
+			_, err = pWallet.IssueAddPermissionlessValidatorTx(
+				&txs.SubnetValidator{Validator: txs.Validator{
```
Comment (abi87) [inline tests/e2e/p/interchain_workflow.go:115]: nit: this is a validator for the primary network. can you explicitly set SubnetID attribute to constants.PrimaryNetworkID?

File: tests/e2e/p/interchain_workflow.go [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{rewardKey.Address()},
+				},
+				delegationFee,
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("adding a delegator to the new node", func() {
```
Comment (abi87) [inline tests/e2e/p/interchain_workflow.go:136]: q: why do we need to make nodeID as both validator and delegator?
Also IIUC, what we are doing here is delegating some more nodeID's funds to nodeID. Why is this needed?

**Review by abi87 — DISMISSED** [review]
a couple of questions about test structure, which may be simpler. Other than that LGTM

**Review by StephenButtolph — APPROVED** [review]
